src/common/transformations/tests/symbolic_transformations/reshape_optimizations.cpp

```diff
@@ -149,3 +149,26 @@ TEST_F(TransformationTestsF, NegativeTest) {
         manager.register_pass<pass::ReshapeOptimizations>();
     }
 }
+
+TEST_F(TransformationTestsF, ZeroDimsInOutputShape) {
+    // [A, B]
+    auto shape = PartialShape{0, 0};
+    {
+        auto data = make_shared<v0::Parameter>(element::f32, shape);
+        auto b = make_shared<v0::Parameter>(element::f32, PartialShape{-1});
+
+        auto a = ov::op::v0::Constant::create(element::i64, Shape{1}, {0});
+
+        auto shape_of = make_shared<v3::ShapeOf>(b);
+        auto indices = ov::op::v0::Constant::create(element::i64, {1}, {0});
+        auto axis = ov::op::v0::Constant::create(element::i64, {}, {0});
+        auto b_dim = make_shared<v1::Gather>(shape_of, indices, axis);
+
+        auto pattern = make_shared<v0::Concat>(OutputVector{a, b_dim}, 0);
+
+        auto reshape = make_shared<v1::Reshape>(data, pattern, false);
+
+        model = make_shared<Model>(NodeVector{reshape}, ParameterVector{data, b});
+        manager.register_pass<pass::ReshapeOptimizations>();
```

Comment:
should we have reference model here?

Reply:
no, if we do not provide a ref model, the model will be compared with its copy.